proxy/http2/Http2ConnectionState.cc

```diff
@@ -139,9 +139,14 @@ rcv_data_frame(Http2ConnectionState &cstate, const Http2Frame &frame)
       return Http2Error(Http2ErrorClass::HTTP2_ERROR_CLASS_CONNECTION, Http2ErrorCode::HTTP2_ERROR_PROTOCOL_ERROR,
                         "recv data bad payload length");
     }
+
+    // Pure END_STREAM
+    if (payload_length == 0) {
+      stream->signal_read_event(VC_EVENT_READ_COMPLETE);
+    }
   }
 
-  // If Data length is 0, do nothing.
+  // If Data length is 0 & no END_STREAM flag, do nothing.
```

Comment:
I found another case of missing `VC_EVENT_READ_COMPLETE` signal on receiving `END_STREAM` flag. In some cases, this also triggers the crash.

Reply:
The comment is confusing because this if statement doesn't check the flag. Combine the if blocks or rephrase the comment.